plugins/InspectionGadgets/src/com/siyeh/ig/style/SizeReplaceableByIsEmptyInspection.java

```diff
@@ -162,28 +155,35 @@ else if (rhs instanceof PsiMethodCallExpression) {
       if (isEmptyCall == null) {
         return;
       }
-      final Object object = ExpressionUtils.computeConstantExpression(flipped ? lhs : rhs);
-      if (!(object instanceof Integer) || ((Integer)object).intValue() != 0) {
+      final Object object = ExpressionUtils.computeConstantExpression(orderedBinaryExpression.getSecondOperand());
+      if (!(object instanceof Integer) && !(object instanceof Long) && !(object instanceof Short)) {
         return;
       }
-      final IElementType tokenType = expression.getOperationTokenType();
-      if (JavaTokenType.EQEQ.equals(tokenType)) {
-        registerError(expression, isEmptyCall, qualifierExpression, referenceName);
-      }
-      if (ignoreNegations) {
-        return;
-      }
-      if (JavaTokenType.NE.equals(tokenType)) {
-        registerError(expression, '!' + isEmptyCall, qualifierExpression, referenceName);
+      int comparedValue = ((Number)object).intValue();
+      if (comparedValue == 0) {
+        final IElementType tokenType = orderedBinaryExpression.getTokenType();
+        if (JavaTokenType.EQEQ.equals(tokenType)) {
+          registerError(expression, isEmptyCall, qualifierExpression, referenceName);
+        }
+        if (ignoreNegations) {
+          return;
+        }
+        if (JavaTokenType.NE.equals(tokenType) || JavaTokenType.GT.equals(tokenType)) {
+          registerError(expression, '!' + isEmptyCall, qualifierExpression, referenceName);
+        }
       }
-      else if (flipped) {
+      else if (comparedValue == 1) {
+        final IElementType tokenType = orderedBinaryExpression.getTokenType();
         if (JavaTokenType.LT.equals(tokenType)) {
+          registerError(expression, isEmptyCall, qualifierExpression, referenceName);
```

Comment:
Do we have any new functionality in this inspection? Or only refactoring?

Reply:
New cases using `1` have been added.